lib/timers.js

```diff
@@ -232,6 +231,12 @@ function listOnTimeout() {
   } else if (list === refedLists[msecs]) {
     delete refedLists[msecs];
   }
+
+  // do not close reused handler which is unrefed in setInterval
+  if (this.owner && this.owner._repeat)
```

Comment:
Is the test on `this.owner._repeat` needed? That is, isn't the test on `this.owner` sufficient?